App/Playlist.js

```diff
@@ -0,0 +1,57 @@
+function Playlist(data) {
+  this.index = data.id;
+  this.playlistName = data.title;
+  this.imageLarge = data.imageLarge;
+  this.imageSmall = data.imageSmall;
+  this.description = data.description;
+  this.songsList = [];
+  this.observers = [];
+}
+Playlist.prototype.setName = function(name){
+  this.playlistName = name;
+};
+Playlist.prototype.addSong = function(song){
+    this.songsList.push(song);
+    this.notify(song);
+};
+Playlist.prototype.register = function(observer) {
+  this.observers.push(observer);
+};
+Playlist.prototype.notify = function(song){
+  for(var i = 0; i < this.observers.length; i++)
```

Comment:
Use an array.forEach (available on all arrays) instead of for loop. Also let the other guys know about it(but use it ONLY when map, reduce, filter doesn't make any sense or cannot be used ).